hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java

```diff
@@ -38,765 +47,986 @@
+
+        final var tokenCreateWrapper =
+                decodeTokenCreateWithoutFees(
+                        decodedArguments.get(0),
+                        true,
+                        decodedArguments.get(1),
+                        decodedArguments.get(2),
+                        aliasResolver);
+        final var fixedFees = decodeFixedFees(decodedArguments.get(3), aliasResolver);
+        final var fractionalFees = decodeFractionalFees(decodedArguments.get(4), aliasResolver);
+        tokenCreateWrapper.setFixedFees(fixedFees);
+        tokenCreateWrapper.setFractionalFees(fractionalFees);
+
+        return tokenCreateWrapper;
+    }
+
+    public TokenCreateWrapper decodeNonFungibleCreate(
+            final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+        final Tuple decodedArguments =
+                decodeFunctionCall(
+                        input,
+                        TOKEN_CREATE_NON_FUNGIBLE_SELECTOR,
+                        TOKEN_CREATE_NON_FUNGIBLE_DECODER);
+
+        return decodeTokenCreateWithoutFees(
+                decodedArguments.get(0), false, BigInteger.ZERO, BigInteger.ZERO, aliasResolver);
+    }
+
+    public TokenCreateWrapper decodeNonFungibleCreateWithFees(
+            final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+        final Tuple decodedArguments =
+                decodeFunctionCall(
+                        input,
+                        TOKEN_CREATE_NON_FUNGIBLE_WITH_FEES_SELECTOR,
+                        TOKEN_CREATE_NON_FUNGIBLE_WITH_FEES_DECODER);
+
+        final var tokenCreateWrapper =
+                decodeTokenCreateWithoutFees(
+                        decodedArguments.get(0),
+                        false,
+                        BigInteger.ZERO,
+                        BigInteger.ZERO,
+                        aliasResolver);
+        final var fixedFees = decodeFixedFees(decodedArguments.get(1), aliasResolver);
+        final var royaltyFees = decodeRoyaltyFees(decodedArguments.get(2), aliasResolver);
+        tokenCreateWrapper.setFixedFees(fixedFees);
+        tokenCreateWrapper.setRoyaltyFees(royaltyFees);
+
+        return tokenCreateWrapper;
+    }
+
+    public TokenInfoWrapper decodeGetTokenInfo(final Bytes input) {
+        final Tuple decodedArguments =
+                decodeFunctionCall(input, GET_TOKEN_INFO_SELECTOR, GET_TOKEN_INFO_DECODER);
+
+        final var tokenID = convertAddressBytesToTokenID(decodedArguments.get(0));
+        return TokenInfoWrapper.forToken(tokenID);
+    }
+
+    public TokenInfoWrapper decodeGetFungibleTokenInfo(final Bytes input) {
+        final Tuple decodedArguments =
+                decodeFunctionCall(
+                        input, GET_FUNGIBLE_TOKEN_INFO_SELECTOR, GET_FUNGIBLE_TOKEN_INFO_DECODER);
+
+        final var tokenID = convertAddressBytesToTokenID(decodedArguments.get(0));
+        return TokenInfoWrapper.forToken(tokenID);
```

Comment:
Q: Why does the return match the exactly match the `decodeGetTokenInfo`?

Reply:
I'm not sure I understand the question correctly. I have defined two methods inside `TokenInfoWrapper`:

- `forToken` - used by both getTokenInfo and `getFungibleTokenInfo` precompiles, since they share the same input information
- `forNonFungibleToken` - used by `getNonFungibleTokenInfo` precompile

Reply:
I see.
My concern was around `forToken` being using for both `getTokenInfo` and `getFungibleTokenInfo`.
`TokenInfo` and `FungibleTokenInfo` are not the same, for the same reason `TokenInfo` and `NonFungibleTokenInfo` are not the same.
Still grasping the architecture here so maybe i'm missing something but seems odd that both `getTokenInfo` and `getFungibleTokenInfo` would use the same call but then `getNonFungibleTokenInfo` wouldn't.
I'd expect them all to be different or all to have a single base call they make with `getFungibleTokenInfo` and `getNonFungibleTokenInfo` having additional logic for the additional params